interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -1,35 +1,23 @@
 /*
  * The volcano-plot module provides a directive and controller for embedding
- * volcano plots in other views. Requires samples to be grouped in the
- * adage.analyze.sampleBin in order to function properly -- if no samples are
- * chosen or placed into groups, an error will be displayed.
+ * volcano plots in other views. Requires properly prepared data to be specified
+ * via plotData.
  */
 angular.module('adage.volcano-plot', [
   'ngVega',
   'greenelab.stats',
-  'adage.analyze.sampleBin',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec', 'SampleBin',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec, SampleBin) {
-    // retrieve the nodeInfo map (for node names) via the SampleBin service
-    // $scope.nodeInfo = SampleBin.getNodeInfo(); // TBD
-    $scope.nodeInfo = [
-      {'id': 1, 'name': 'Node1pos'},
-      {'id': 2, 'name': 'Node2pos'},
-      {'id': 518, 'name': 'test name'}
-    ];
-
+.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
+  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
     // define controller instance variables that link with vega
-    this.data = {
-      source: $scope.plotData
-      // FIXME: this is not working as it should to bring in node names
-      // nodes: $scope.nodeInfo
-      // FIXME: can't get selectedNodes out this way... need to use view api?
-      // selectedNodes: $scope.selectedNodes
-    };
     this.spec = VolcanoPlotSpec;
+    this.data = $scope.plotData;
+    // FIXME: this is not working as it should to bring in node names
+    // $scope.plotData.nodes: $scope.nodeInfo
+    // FIXME: can't get selectedNodes out this way... need to use view api?
+    // $scope.plotData.selectedNodes: $scope.selectedNodes
```

Comment:
I feel similarly about these FIXMEs. I know you didn't write them (and since writing the other comment about the FIXME I realized that you only changed the indentation there as well), but are they ever going to be fixed or are they just cluttering up the code at this point? Oh wait, it seems like this PR may actually be fixing at least one of these.